fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateTableCommand.java

```diff
@@ -0,0 +1,166 @@
+
+import org.apache.doris.analysis.StmtType;
+import org.apache.doris.catalog.Column;
+import org.apache.doris.catalog.DatabaseIf;
+import org.apache.doris.catalog.Env;
+import org.apache.doris.catalog.ScalarType;
+import org.apache.doris.catalog.Table;
+import org.apache.doris.catalog.TableIf;
+import org.apache.doris.catalog.View;
+import org.apache.doris.datasource.hive.HMSExternalTable;
+import org.apache.doris.datasource.hive.HiveMetaStoreClientHelper;
+import org.apache.doris.nereids.trees.plans.PlanType;
+import org.apache.doris.nereids.trees.plans.commands.info.TableNameInfo;
+import org.apache.doris.nereids.trees.plans.visitor.PlanVisitor;
+import org.apache.doris.qe.ConnectContext;
+import org.apache.doris.qe.ShowResultSet;
+import org.apache.doris.qe.ShowResultSetMetaData;
+import org.apache.doris.qe.StmtExecutor;
+
+import com.google.common.collect.Lists;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Represents the command for SHOW CREATE TABLE.
+ */
+public class ShowCreateTableCommand extends Command implements NoForward {
+    private static final ShowResultSetMetaData META_DATA =
+            ShowResultSetMetaData.builder()
+            .addColumn(new Column("Table", ScalarType.createVarchar(20)))
+            .addColumn(new Column("Create Table", ScalarType.createVarchar(30)))
+            .build();
+
+    private static final ShowResultSetMetaData VIEW_META_DATA =
+            ShowResultSetMetaData.builder()
+            .addColumn(new Column("View", ScalarType.createVarchar(20)))
+            .addColumn(new Column("Create View", ScalarType.createVarchar(30)))
+            .addColumn(new Column("character_set_client", ScalarType.createVarchar(30)))
+            .addColumn(new Column("collation_connection", ScalarType.createVarchar(30)))
+            .build();
+
+    private static final ShowResultSetMetaData MATERIALIZED_VIEW_META_DATA =
+            ShowResultSetMetaData.builder()
+            .addColumn(new Column("Materialized View", ScalarType.createVarchar(20)))
+            .addColumn(new Column("Create Materialized View", ScalarType.createVarchar(30)))
+            .build();
+
+    private final String catalog;
+    private final String dbName;
+    private final String tableName;
+    private final boolean isBrief;
+
+    public ShowCreateTableCommand(TableNameInfo tableNameInfo, boolean isBrief) {
+        super(PlanType.SHOW_CREATE_TABLE_COMMAND);
+        this.catalog = tableNameInfo.getCtl();
+        this.dbName = tableNameInfo.getDb();
+        this.tableName = tableNameInfo.getTbl();
+        this.isBrief = isBrief;
+    }
+
+    public String getCatalog() {
+        return catalog;
+    }
+
+    public String getDbName() {
+        return dbName;
+    }
+
+    public String getTableName() {
+        return tableName;
+    }
+
+    public boolean isBrief() {
+        return isBrief;
+    }
+
+    @Override
+    public void run(ConnectContext ctx, StmtExecutor executor) throws Exception {
```

Comment:
ShowCreateTableStmt's analyze method needs to be re-implemented in nereids. For example, add a `private void validate(ConnectContext ctx) throws UserException` method to do the job, and call it at the beginning of run method.

Reply:
Done - please review